### ISSUE TITLE
ConnectionRefused: Cannot create app offline or under corporate proxy / firewall

### ISSUE THREAD

**Issue by polRk** [oven-sh/bun#5722]
### What version of Bun is running?

1.0.2+37edd5a6e389265738e89265bcbdf2999cb81a49

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

Run `bun create elysia hi-elysia`

### What is the expected behavior?

Fetching package.json....

### What do you see instead?

Install offline or support private/proxy npm registry (for example nexus) 

### Additional information

All Internet resources except corporate ones are closed

**Comment by polRk** [comment]
```

error: ConnectionRefused

----- bun meta -----
Bun v1.0.2 (37edd5a6) Linux x64 #1 SMP Thu May 18 10:27:05 EDT 2023
CreateCommand: 
Elapsed: 130615ms | User: 4ms | Sys: 4ms
RSS: 36.71MB | Peak: 16.67MB | Commit: 36.71MB | Faults: 0
----- bun meta -----

0   0x55b0879d958b
1   ???
2   ???
3   ???
4   ???
5   ???

Search GitHub issues https://bun.sh/issues or ask for #help in https://bun.sh/discord
```

**Comment by gildembergleite** [comment]
I have this same issue

**Comment by coopbri** [comment]
Hey, similar thing (can't install dependencies, `ConnectionRefused downloading package manifest`) on personal VPN. Works fine with it disabled, works fine with `npm`, `yarn`, and `pnpm`, and same issue regardless of IPv4-only or v6-enhanced tunneling.

**Comment by kikindb** [comment]
I have the same error and I'm using my regular internet, no vpn, proxy, etc. My OS is Ubuntu 23.10

**Comment by anru** [comment]
@polRk I think the nature of this issue described in bug #11582 and should be fixed in next release.

**Comment by Jarred-Sumner** [comment]
@anru can you run `bun upgrade --canary` and verify the issues you’ve been running into are fixed? We’ll do the release later today

**Comment by anru** [comment]
@Jarred-Sumner Yes, I do confirm that version 1.1.13-canary.1 does not have the above issues (including success result for `bun create elysia hi-elysia` command).

Although I did encounter a new problem when trying to install packages in one of the repositories, but it may have other causes:

```
❯ bun add -g lightningcss-win32-x64-msvc@1.25.1
bun add v1.1.13-canary.1 (386bc212)

installed lightningcss-win32-x64-msvc@1.25.1

[1011.00ms] done

❯ bun install
bun install v1.1.13-canary.1 (386bc212)
error: lightningcss-win32-x64-msvc@1.25.1 failed to resolve
```

This seems a bit odd - the package is installed globally, but cannot be resolved when installed as a project dependency.
After that I removed the installation caches: `rm -rf ~/.bun/install/cache` and ran the command again: after that the error above is gone.

Ok, I ran the command again, and ended up getting the following error:

<img width="823" alt="image" src="https://github.com/oven-sh/bun/assets/130013/8d255eed-021b-41d0-a200-133271e0a1aa">

Looks like bun don't link all binaries before trying run package installation scripts.

Oookay. I ran `bun install` again. 

```
bun install v1.1.13-canary.1 (386bc212)

255 packages installed [5.16s]
```

And finally, it works!
